doc/api/http.md

```diff
@@ -2161,6 +2161,25 @@ In the case of a connection error, the following events will be emitted:
 * `'error'`
 * `'close'`
 
+In the case of a premature connection close before the response is received,
+the following events will be emitted in the following order:
+
+* `'socket'`
+* `'error'` with an error with message `'Error: socket hang up'` and code
```

Comment:
I'm not sure the actual error message here is useful/relevant as long as the `code` is mentioned.